GitHubActionsTestLogger/CliOptionsProvider.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Testing.Platform.CommandLine;
+using Microsoft.Testing.Platform.Extensions;
+using Microsoft.Testing.Platform.Extensions.CommandLine;
+
+namespace GitHubActionsTestLogger;
+
+internal sealed class CliOptionsProvider(GitHubTestReporterExtension extension) : ICommandLineOptionsProvider
+{
+    public const string ReportGitHubOption = "report-github";
+    public const string ReportGitHubSummaryOption = "report-github-summary";
+    public const string ReportGitHubTitleOption = "report-github-title";
+    public const string ReportGitHubMessageOption = "report-github-message";
+
+    public static class ReportGitHubSummaryArguments
+    {
+        public const string IncludePassedTests = "includePassedTests";
+        public const string IncludeSkippedTests = "includeSkippedTests";
+        public const string IncludeNotFoundTests = "includeNotFoundTests";
+    }
+
+    public string Uid => extension.Uid;
+    public string Version => extension.Version;
+    public string DisplayName => extension.DisplayName;
+    public string Description => extension.Description;
+
+    public IReadOnlyCollection<CommandLineOption> GetCommandLineOptions() =>
+        [
+            new(ReportGitHubOption, "Reports test run information to GitHub Actions", ArgumentArity.Zero, isHidden: false),
+            // TODO: Do you prefer multiple zero argument options or a single option with argument?
```

Comment:
If it were me, I'd probably prefer individual ones as then when consuming via MSBuild it would be conceptually one-to-one when building up the arguments to pass through with `TestingPlatformCommandLineArguments` so conditionally deciding what to turn on/off is simpler.

Reply:
So with the current implementation, would this option be used like so?

```
... --report-github-summary includePassedTests includeSkippedTests
```

In that case, I feel like it'd be more user friendly and idiomatic to have a separate option for each flag:

```
--report-github-summary-include-passed-tests
--report-github-summary-include-skipped-tests
```

It's kind of a mouthful though, which is why I opted for the "nested" convention in the current version in the first place.